src/core/lib/surface/legacy_channel.cc

```diff
@@ -97,9 +97,23 @@ absl::StatusOr<OrphanablePtr<Channel>> LegacyChannel::Create(
                                 .value_or(CoreConfiguration::Get()
                                               .resolver_registry()
                                               .GetDefaultAuthority(target));
+    experimental::StatsPluginChannelScope scope(target, authority);
     *(*r)->stats_plugin_group =
-        GlobalStatsPluginRegistry::GetStatsPluginsForChannel(
-            experimental::StatsPluginChannelScope(target, authority));
+        GlobalStatsPluginRegistry::GetStatsPluginsForChannel(scope);
+    // Add per-channel stats plugins.
+    auto* stats_plugin_list = args.GetPointer<
+        std::shared_ptr<std::vector<std::shared_ptr<StatsPlugin>>>>(
+        GRPC_ARG_STATS_PLUGINS);
+    if (stats_plugin_list != nullptr) {
+      for (const auto& plugin : **stats_plugin_list) {
+        bool is_enabled = false;
+        std::shared_ptr<StatsPlugin::ScopeConfig> config;
+        std::tie(is_enabled, config) = plugin->IsEnabledForChannel(scope);
```

Comment:
This doesn't seem like quite the right interface here.  In this case, we are being explicitly told to use this stats plugin for this channel, so we shouldn't ask the plugin if it's okay and give it the opportunity to say no.  Instead, I think we should have a method on the plugin to get the config for the channel unconditionally.